File: fullEcommerce/models/order.js
const mongoose = require("mongoose")
const Product = require("./Product")
const date = new Date()
const order = mongoose.model("order",{
User:{
    type:mongoose.Schema.Types.ObjectId,
    ref:"User"
},
Product:[
    {

            product:{
                type:mongoose.Schema.Types.ObjectId,
                ref:"Product"
            },
            qty:{
                type:Number
            }

    }
],
PhoneNumber:{
    type:String,
    require:true
},
Latitude:{
    type:String,
    default:""
}
,
Longtitude:{
    type:String,
    default:""

},
Total:{
    type:Number,
    default:0

}
,
orderStatus: {// delivered // cancelled // failed
    type: Boolean,
    default:false
  },
  DeliveredStatus:{
    type:Boolean,
    default:false
  },
  EmployeeId:{

    type:mongoose.Schema.Types.ObjectId,
    ref:'User',
    deault:""
  },
  Address:{
      type:String,
    default:""

  },
  Date:{
      type:String,
      default:`${date.getFullYear()}/${date.getMonth()}/${date.getDay()}`
  }



})

module.exports = order

File: app/src/main/java/com/xrest/shopify/Retrofit/Class/Order.kt
package com.xrest.shopify.Retrofit.Class

import java.io.Serializable

data class Order(
        var  _id:String?=null,
        var User:User?=null,
        var Product:MutableList<BP>?=null,
        var PhoneNumber:String?=null,
        var Longtitude:String?=null,
        var Latitude:String?=null,
        var Total:Int?=null,
        var orderStatus:Boolean?=null,
        var DeliveredStatus:Boolean?=null,
        var EmployeeId:User?=null,
        var Address:String?=null,
        var Date:String?=null
):Serializable
